richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/model/RichTextState.kt

```diff
@@ -569,21 +583,41 @@ class RichTextState internal constructor(
         )
     }
 
+    @Deprecated("Use toggleUnorderedList() instead to add or remove unordered list")
     fun removeUnorderedList() {
         val paragraph = getRichParagraphByTextIndex(selection.min - 1) ?: return
+        removeUnorderedList(paragraph)
+    }
+
+    private fun removeUnorderedList(paragraph: RichParagraph) {
         if (paragraph.type !is UnorderedList) return
 
         resetParagraphType(paragraph = paragraph)
     }
 
     fun toggleOrderedList() {
-        val paragraph = getRichParagraphByTextIndex(selection.min - 1) ?: return
-        if (paragraph.type is OrderedList) removeOrderedList()
-        else addOrderedList()
+        val firstParagraph = getRichParagraphByTextIndex(selection.min - 1) ?: return
```

Comment:
Why you are getting the first paragraph here, the paragraph list should be enough.
And if the list if empty, you can return

Reply:
Thank you for the tip!

When I tested the new version, I've found out, that when I tap on text only, without any selection,  list styles doesn't applied to current paragraph. It occures because getRichParagraphListByTextRange() returns empty list. So, I've edited getRichParagraphListByTextRange() for this case. 
If it's not good for the main concept, I can create separate function

Reply:
Yes, it's fine now. The `getRichParagraphListByTextRange` method should return data even if the selection is collapsed.